external/concealer/include/concealer/publisher.hpp

```diff
@@ -37,69 +39,138 @@ struct Identity
   }
 };
 
+template <typename ValueType>
+struct NormalDistributionError
+{
+  static_assert(
+    std::disjunction_v<
+      std::is_same<std::decay_t<ValueType>, float>, std::is_same<std::decay_t<ValueType>, double>,
+      std::is_same<std::decay_t<ValueType>, long double> >,
+    "Unsupported error type");
+
+  /// @note set this to false to disable randomization
+  bool active{true};
+
+  std::normal_distribution<ValueType> additive, multiplicative;
+
+  explicit NormalDistributionError(
+    const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr & node,
+    const std::string & prefix)
+  // clang-format off
+  : additive(
+      static_cast<ValueType>(common::getParameter<double>(node, prefix + ".additive.mean")),
+      static_cast<ValueType>(common::getParameter<double>(node, prefix + ".additive.standard_deviation"))),
+    multiplicative(
+      static_cast<ValueType>(common::getParameter<double>(node, prefix + ".multiplicative.mean")),
+      static_cast<ValueType>(common::getParameter<double>(node, prefix + ".multiplicative.standard_deviation")))
+  // clang-format on
+  {
+  }
+
+  auto apply(std::mt19937_64 & engine, const ValueType value) -> ValueType
+  {
+    if (active) {
+      return value * (multiplicative(engine) + static_cast<ValueType>(1)) + additive(engine);
+    } else {
+      return value;
+    }
+  }
+};
+
 template <typename>
 struct NormalDistribution;
 
+/**
+ * @brief Provides common components for obtaining the seed and initializing the pseudo random number generator engine
+ * obtains the seed from the parameter <topic>.seed
+ * initializes `engine` appropriately
+ */
+struct NormalDistributionBase
```

Comment:
The name is inappropriate because (as you note in your comment) this structure is not just for normal distributions.

```suggestion
struct RandomNumberEngine
```

Reply:
Done